README.rst

```diff
@@ -0,0 +1,235 @@
+###################
+pypyr slack plug-in
+###################
+
+.. image:: https://cdn.345.systems/wp-content/uploads/2017/03/pypyr-logo-small.png
+    :alt: pypyr-logo
+    :align: left
+
+*pypyr*
+    pronounce how you like, but I generally say *piper* as in "piping down the
+    valleys wild"
+
+Send messages to `slack <https://slack.com/>`__ from pypyr. This is useful for
+sending notifications on success or failure conditions in your pipelines. Or
+for sending a message just because you can.
+
+`pypyr <https://github.com/pypyr/pypyr-cli>`__ is a command line interface to
+run pipelines defined in yaml.
+
+|build-status| |coverage| |pypi|
+
+.. contents::
+
+.. section-numbering::
+
+************
+Installation
+************
+
+pip
+===
+.. code-block:: bash
+
+  # pip install --upgrade pypyr-slack
+
+pypyr-slack depends on pypyr-cli. The above ``pip`` will install it for you if
+you don't have it already.
+
+Python version
+==============
+Tested against Python 3.6
+
+*****
+steps
+*****
+pypyrslack.steps.send
+=====================
+Send a message to slack.
+
+Required Context
+----------------
+
+Requires the following context items:
+
+- slackToken
+
+  - your slack api token. Keep this secure.
+- slackChannel
+
+  - send to this slack channel (include # in front)
+- slackText
+
+  - the body of your message. Use your usual slack formatting chars.
+
+Text substitutions
+------------------
+For both slackChannel and slackText you can use substitution tokens, or string
+interpolation. This substitutes anything between curly braces with the context
+value for that key. For example, if your context looked like this:
+
+.. code-block:: yaml
+
+  arbitraryValue: pypyrchannel
+  arbitraryText: down the
+  moreArbText: wild
+  slackChannel: "#{arbitraryValue}"
+  slackText: "piping {arbitraryText} the valleys {moreArbText}"
```

Comment:
William Blake. I'll keep the spelling to piping in his honour.

Yes, the output text would be "piping down the valleys wild"